src/ansys/units/dimensions.py

```diff
@@ -1,142 +1,100 @@
-                raise DimensionsError.EXCESSIVE_DIMENSIONS(len(dimensions))
-
-            self._dimensions, self._unit = self._dim_to_units(
-                dimensions=dimensions, unit_sys=unit_sys
-            )
-
-    def _dim_to_units(self, dimensions: list, unit_sys: list) -> str:
-        """
-        Convert a dimensions list into a unit string.
-
-        Parameters
-        ----------
-        dimensions : list
-            List of unit dimensions.
-
-        unit_sys : list
-            Unit system of the dimensions.
-
-        Returns
-        -------
-        str
-            Unit string representation of the dimensions.
-        """
-        # Ensure dimensions list contains 9 terms
-        dimensions = [
-            float(dim)
-            for dim in dimensions + ((self.max_dim_len() - len(dimensions)) * [0])
-        ]
-        units = ""
-
-        # Define unit term and associated value from dimension with dimensions list
-        for idx, dim in enumerate(dimensions):
-            if dim == 1.0:
-                units += f" {unit_sys[idx]}"
-            elif dim != 0.0:
-                dim = int(dim) if dim % 1 == 0 else dim
-                units += f" {unit_sys[idx]}^{dim}"
-
-        return dimensions, condense(units=units)
-
-    def _units_to_dim(
-        self, units: str, power: float = None, dimensions: list = None
-    ) -> list:
-        """
-        Convert a unit string into a dimensions list.
-
-        Parameters
-        ----------
-        units : str
-            Unit string of quantity.
-        power : float
-            Power of unit string.
-
-        Returns
-        -------
-        list
-            Dimensions representation of unit string.
-        """
-        power = power or 1.0
-        dimensions = dimensions or [0.0] * self.max_dim_len()
-
-        # Split unit string into terms and parse data associated with individual terms
-        for term in units.split(" "):
-            _, unit_term, unit_term_power = filter_unit_term(term)
-
-            unit_term_power *= power
-
-            # retrieve data associated with fundamental unit
-            if unit_term in ansunits._fundamental_units:
-                idx = (
-                    ansunits._dimension_order[
-                        ansunits._fundamental_units[unit_term]["type"]
-                    ]
-                    - 1
-                )
-                dimensions[idx] += unit_term_power
-
-            # Retrieve derived unit composition unit string and factor.
-            if unit_term in ansunits._derived_units:
-                # Recursively parse composition unit string
-                dimensions = self._units_to_dim(
-                    units=ansunits._derived_units[unit_term]["composition"],
-                    power=unit_term_power,
-                    dimensions=dimensions,
-                )
-
-        return dimensions
+    def __init__(self, dimensions_container: (dict[str, float], list[float]) = None):
```

Comment:
Since the order of dimensions is important here should we force API users to use this:

https://docs.python.org/3/library/collections.html#collections.OrderedDict

?
If I construct the unordered dictionary manually in my code I don't think there is any guarantee that the keys come in the order that I wrote them down.

Reply:
I am a bit confused on this. Would you be able to give me an example?

Reply:
For our purposes, we just need a dict-like object that maps the dimension ids to powers. The OrderedDict is only remembering the arbitrary insertion order and that isn't really any usable additional information.

Reply:
OrderedDict only makes sense if the insertion order to the dictionary is not arbitrary, I think in the previous code it could not be, but you are right, with the Dimensions enum now it does not matter.